harvester/harvester
# Storage network setting controller

**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**

https://github.com/harvester/harvester/issues/1055

**Solution:**

controller will shutdown Pods with longhorn volume and check all volumes are detached before applying LH storage network config

**Related Issue:**
https://github.com/harvester/harvester/issues/1055

**TODO**

Need to update webhook to verify Value and VM status

**Test plan:**


## Test1

1. Shutdown all VMs
2. setup vlan config & cluster network
3. set settings named "storage-network" with value below in Harvester settings
```json
{
    "vlan": 1,
    "clusterNetwork": "<your cluster network>",
    "range": "<IP range>"
}
```
6. wait and check Longhorn settings "storage-network" is "storagenetwork-XXXXXXXXXXX".
7. check Longhorn `instance-manager-e-XXXX` has an interface named "lhnet1"

## Test2

1. do Test1 and wait
2. set value=""
3. wait and check Longhorn "storage-network" is empty
4. check Longhoorn `instance-manager-e-XXXX` didn't has an interface name "lhnet1"
